src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/vmware/VMWareProvider.java

```diff
@@ -231,12 +229,10 @@ public void deleteInstance(Infrastructure infrastructure, String instanceId) {
                                         .findFirst()
                                         .ifPresent(vm -> {
                                             try {
-
                                                 if (Task.SUCCESS.equals(vm.powerOffVM_Task().waitForTask())) {
                                                     Task task = vm.destroy_Task();
                                                     String result = task.waitForTask();
                                                     if (!result.equals(Task.SUCCESS)) {
-
                                                         throw new RuntimeException("Unable to delete VMWare instance : " +
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS00112)